lib/spack/spack/binary_distribution.py

```diff
@@ -300,7 +301,10 @@ def build_tarball(spec, outdir, force=False, rel=False, unsigned=False,
         if force:
             os.remove(spackfile_path)
         else:
-            raise NoOverwriteException(str(spackfile_path))
+            err_msg = "\n%s\nexists\n" % spackfile_path
+            err_msg += "Use -f option to overwrite."
+            tty.warn(err_msg)
+            return
```

Comment:
What's the rationale behind demoting this exception to a warning? Also https://github.com/FNALssi/spack/blob/5b31a6ddc7ee992038bb71e2b6ab9247c4b4c13c/lib/spack/spack/binary_distribution.py#L320-L322

Reply:
Convenience. I keep meaning to make this change upstream. I only remember when create buildcaches.

Reply:
Not sure what you mean by convenience, here: it's convenient for it not to be an exception because...? Generally if the final product would be non-functional and there's no way to recover automatically, it should be an exception.

Reply:
I put back the exception. The problem is that is you specify multiple packages which share dependencies then the second one fails.